code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm

```diff
@@ -75,7 +75,10 @@
 		if (length(exclusive_roles) && !(creature.mind.assigned_role.title in exclusive_roles)) // Is the rule exclusive to their job?
 			trimmed_list.Remove(creature)
 			continue
-		if(HAS_TRAIT(creature, TRAIT_MIND_TEMPORARILY_GONE)) // are they in the vdom?
+		if(HAS_TRAIT(creature, TRAIT_MIND_TEMPORARILY_GONE)) // are they out of body?
+			trimmed_list.Remove(creature)
+			continue
+		if(!isnull(creature.GetComponent(/datum/component/avatar_connection))) // are they an avatar?
```

Comment:
This should be a trait of its own, `TRAIT_TEMPORARY_BODY`, no getcomponent + I can see it getting some other use (cult ghosts for example)

Reply:
added `TRAIT_TEMPORARY_BODY`